codebuild/bin/s2n_open_fds_test.py

```diff
@@ -0,0 +1,66 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script parses the LastDynamicAnalysis file generated by CTest Memcheck.
+# It identifies any leaking file descriptors and triggers an error when detected.
+# This enhances the capabilities of existing Valgrind checks.
+import os
+import re
+import sys
+
+# Exit code matches the valgrind memory loss error code
+ERROR_EXIT_CODE = 123
+
+analysis_file_pattern = re.compile(r"LastDynamicAnalysis.*")
```

Comment:
1. In situation like this I always like to include a concrete example of the thing that we are matching
2. Could we match it in a simpler way? E.g. is it enough to just select the lexicographically largest `LastDynamicAnalysis`? Generally I treat regex's as a matter of last resort. We should use the simplest solution that we can, and regex's are pretty complicated.

Reply:
> In situation like this I always like to include a concrete example of the thing that we are matching.

I have a regex test script which shows what exactly each regex is matching for.
> Is it enough to just select the lexicographically largest LastDynamicAnalysis?

Yes, I think that is reasonable. Python `in` is very helpful and can replace all regex usages. I will do it this way, and remove all regexes.